packages/snap-client/src/Client/NetworkCache/NetworkCache.test.ts

```diff
@@ -79,21 +119,247 @@ describe('Network Cache', () => {
 		expect(fetchfn).toHaveBeenCalledTimes(2);
 
 		//now we have cache
-		expect(mockStorage['ss-networkcache']).toBeDefined();
+		expect(mockStorage[CACHE_STORAGE_KEY]).toBeDefined();
 
 		// we used the cache
 		expect(global.Storage.prototype.getItem).toHaveBeenCalled();
 
+		controller.urlManager.reset().set('query', 'dress').go();
 		//make another call
 		await controller.search();
 
 		//cache was updated
-		expect(mockStorage['ss-networkcache']).toBeDefined();
+		expect(mockStorage[CACHE_STORAGE_KEY]).toBeDefined();
 
-		//but it did not make additional calls and used previous cache response
-		expect(fetchfn).toHaveBeenCalledTimes(2);
+		//it did make additional calls because the params changed
+		expect(fetchfn).toHaveBeenCalledTimes(4);
 
 		//check that there are results that we pulled from cache
 		expect(controller.store.results.length).toBeGreaterThan(0);
+
+		controller.urlManager.reset().set('query', '').go();
+		//make another call
+		await controller.search();
+
+		//but it did not make additional calls and used previous cache response
+		expect(fetchfn).toHaveBeenCalledTimes(4);
+
+		fetchfn.mockClear();
+
+		fetchfn.mockReset();
+	});
+
+	it('can be disabled via the config', async () => {
+		const clientConfig = {
+			search: {
+				cache: {
+					enabled: false,
+				},
+			},
+			meta: {
+				cache: {
+					enabled: false,
+				},
+			},
+		};
+
+		const controller = new SearchController(searchConfig, {
+			client: new Client(globals, clientConfig),
+			store: new SearchStore(searchConfig, services),
+			urlManager,
+			eventManager: new EventManager(),
+			profiler: new Profiler(),
+			logger: new Logger(),
+			tracker: new Tracker(globals),
+		});
```

Comment:
This is a good test. However, it is more of an integration test than a unit test - if anything it should belong in the Client test.

Reply:
There is quite a bit of boilerplate and set up needed for this test. All of the mock storage set up and imports needed for the search controller. Do we want to move all of that over to the client test? The mock storage bits would be duplicated. Since they are also used in other tests in this file.

Reply:
I think unit tests should only be testing the functionality provided by the `NetworkCache` class - the constructor, and the methods.

In these tests you have setup the controller for: `caches search responses and uses them` and `can be disabled via the config` where you are testing more than the `NetworkCache` class. For example, the disable config test could just pass a config to the constructor that disables the cache without the need for the controller and all of its services. Testing that the client config passes the config to the `NetworkCache` is more of a `Client` or integration test.

These are good tests, but in my opinion do not belong here. I'd suggest putting these in an "integration" test (similar to the [UrlManager](https://github.com/searchspring/snap/blob/main/packages/snap-url-manager/src/integration.test.ts)) - where the entire setup can be tested all together. If you do that, the `metaResponse` used in the mocking should probably be moved to a `MockData` entry.